File: app/build 4/tmp/kapt3/stubs/debug/com/okankkl/movieapp/di/AppModule.java
package com.okankkl.movieapp.di;

@dagger.Module
@kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u00004\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0003\b\u0007\u0018\u00002\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0002J\u0012\u0010\u0003\u001a\u00020\u00042\b\b\u0001\u0010\u0005\u001a\u00020\u0006H\u0007J\u0012\u0010\u0007\u001a\u00020\b2\b\b\u0001\u0010\u0005\u001a\u00020\u0006H\u0007J\u0018\u0010\t\u001a\u00020\n2\u0006\u0010\u000b\u001a\u00020\f2\u0006\u0010\r\u001a\u00020\bH\u0007J\u0010\u0010\u000e\u001a\u00020\u000f2\u0006\u0010\u0010\u001a\u00020\u0004H\u0007J\b\u0010\u0011\u001a\u00020\fH\u0007\u00a8\u0006\u0012"}, d2 = {"Lcom/okankkl/movieapp/di/AppModule;", "", "()V", "provideAppPreference", "Lcom/okankkl/movieapp/data/local/dataStore/AppPreference;", "context", "Landroid/content/Context;", "provideMovieDao", "Lcom/okankkl/movieapp/data/local/room/database/MovieDao;", "provideMovieRepository", "Lcom/okankkl/movieapp/domain/repository/MovieRepository;", "movieApi", "Lcom/okankkl/movieapp/data/remote/MovieApi;", "movieDao", "providePreferenceRepository", "Lcom/okankkl/movieapp/domain/repository/PreferenceRepository;", "appPreference", "provideRetrofit", "app_debug"})
@dagger.hilt.InstallIn(value = {dagger.hilt.components.SingletonComponent.class})
public final class AppModule {
    
    public AppModule() {
        super();
    }
    
    @javax.inject.Singleton
    @dagger.Provides
    @org.jetbrains.annotations.NotNull
    public final com.okankkl.movieapp.data.remote.MovieApi provideRetrofit() {
        return null;
    }
    
    @javax.inject.Singleton
    @dagger.Provides
    @org.jetbrains.annotations.NotNull
    public final com.okankkl.movieapp.data.local.room.database.MovieDao provideMovieDao(@dagger.hilt.android.qualifiers.ApplicationContext
    @org.jetbrains.annotations.NotNull
    android.content.Context context) {
        return null;
    }
    
    @javax.inject.Singleton
    @dagger.Provides
    @org.jetbrains.annotations.NotNull
    public final com.okankkl.movieapp.domain.repository.MovieRepository provideMovieRepository(@org.jetbrains.annotations.NotNull
    com.okankkl.movieapp.data.remote.MovieApi movieApi, @org.jetbrains.annotations.NotNull
    com.okankkl.movieapp.data.local.room.database.MovieDao movieDao) {
        return null;
    }
    
    @javax.inject.Singleton
    @dagger.Provides
    @org.jetbrains.annotations.NotNull
    public final com.okankkl.movieapp.data.local.dataStore.AppPreference provideAppPreference(@dagger.hilt.android.qualifiers.ApplicationContext
    @org.jetbrains.annotations.NotNull
    android.content.Context context) {
        return null;
    }
    
    @javax.inject.Singleton
    @dagger.Provides
    @org.jetbrains.annotations.NotNull
    public final com.okankkl.movieapp.domain.repository.PreferenceRepository providePreferenceRepository(@org.jetbrains.annotations.NotNull
    com.okankkl.movieapp.data.local.dataStore.AppPreference appPreference) {
        return null;
    }
}

File: app/src/main/java/com/okankkl/movieapp/di/AppModule.kt
package com.okankkl.movieapp.di

import android.content.Context
import com.okankkl.movieapp.data.data_source.DataStoreManager
import com.okankkl.movieapp.data.data_source.LocalDataSource
import com.okankkl.movieapp.data.data_source.RemoteDataSource
import com.okankkl.movieapp.data.local.dataStore.AppPreference
import com.okankkl.movieapp.data.local.room.MovieDao
import com.okankkl.movieapp.data.local.room.MovieDatabase
import com.okankkl.movieapp.data.remote.MovieApi
import com.okankkl.movieapp.data.repository.MovieRepositoryImp
import com.okankkl.movieapp.data.repository.PreferenceRepositoryImp
import com.okankkl.movieapp.data.repository.MovieRepository
import com.okankkl.movieapp.data.repository.PreferenceRepository
import com.okankkl.movieapp.util.Constants
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Singleton

@InstallIn(SingletonComponent::class)
@Module
class AppModule
{
    @Singleton
    @Provides
    fun provideRetrofit() : MovieApi {
        return Retrofit.Builder()
            .baseUrl(Constants.API_BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(MovieApi::class.java)
    }
    @Singleton
    @Provides
    fun provideMovieDao(@ApplicationContext context : Context) : MovieDao
    {
        return MovieDatabase
            .getDatabase(context)
            .dao()
    }
    @Singleton
    @Provides
    fun provideAppPreference(@ApplicationContext context: Context) : AppPreference
            = AppPreference(context)
    
    @Singleton
    @Provides
    fun provideDataStoreManager(appPreference: AppPreference) : DataStoreManager
            = DataStoreManager(appPreference)
    
    @Singleton
    @Provides
    fun provideLocalDataSource(movieDao: MovieDao) : LocalDataSource
            = LocalDataSource(movieDao)
    
    @Singleton
    @Provides
    fun provideRemoteDataSource(movieApi: MovieApi) : RemoteDataSource
            = RemoteDataSource(movieApi)
    
    @Singleton
    @Provides
    fun provideMovieRepository(dataStoreManager: DataStoreManager,localDataSource: LocalDataSource,
                               remoteDataSource: RemoteDataSource) : MovieRepository
    = MovieRepositoryImp(dataStoreManager,localDataSource,remoteDataSource)
    
    @Singleton
    @Provides
    fun provideAppPreferenceRepository(appPreference: AppPreference) : PreferenceRepository{
        return PreferenceRepositoryImp(appPreference)
    }
    
}